test/scale_collection_test.cpp

```diff
@@ -233,3 +337,22 @@ TEST(Scale, DISABLED_encodeVeryLongCollectionUint8) {
 
   ASSERT_EQ(stream.hasMore(1), false);
 }
+
+/**
+ * @given map of <uint32_t, uint32_t>
+ * @when encodeCollection is applied
+ * @then expected result is obtained: header is 2 byte, items are pairs of 4
+ * byte elements each
+ */
+TEST(Scale, encodeMapTest) {
+  std::map<uint32_t, uint32_t> collection = {{1, 5}, {2, 6}, {3, 7}, {4, 8}};
+  ScaleEncoderStream s;
+  ASSERT_NO_THROW((s << collection));
+  auto &&out = s.to_vector();
+
+  auto stream = ScaleDecoderStream(gsl::make_span(out));
+  std::map<uint32_t, uint32_t> decoded;
+  stream >> decoded;
+  ASSERT_TRUE(std::equal(
+      decoded.begin(), decoded.end(), collection.begin(), collection.end()));
+}
```

Comment:
pls add line break to the end.
IDE can be configured to do so automatically on save